packages/gatsby-plugin-offline/src/gatsby-browser.js

```diff
@@ -40,8 +39,14 @@ exports.onServiceWorkerActive = ({
     )
   )
 
-  serviceWorker.active.postMessage({
-    api: `gatsby-runtime-cache`,
-    resources: [...resources, ...prefetchedResources],
+  const resources = [...headerResources, ...prefetchedResources]
+  resources.forEach(resource => {
+    // Create a prefetch link for each resource, so Workbox runtime-caches them
+    const link = document.createElement(`link`)
+    link.rel = `prefetch`
+    link.href = resource
+
+    document.head.appendChild(link)
```

Comment:
Should the order of these (append vs. onload) be swapped? I'd think we'd want to register the onload/remove behavior _before_ adding it. If nothing else, seems to be more intuitive that way, even if both work.

Reply:
These can be reordered if you like - although it should work fine as-is, since the browser doesn't start fetching until it's appended. Would you like me to swap the order?

Reply:
Sure, just for clarity! More of a nit than substantive, but I think it's a bit clearer.